modules/gdnative/android/android_gdn.cpp

```diff
@@ -68,6 +65,33 @@ jobject GDAPI godot_android_get_activity() {
 #endif
 }
 
+jobject GDAPI godot_android_get_surface() {
+#ifdef __ANDROID__
+	// TODO implement! Should somehow obtain this from our GodotView instance (or is this our GodotView instance?)
```

Comment:
You can get the `surface` from the `GodotView` instance using `GodotView.getHolder().getSurface()`.

You can either use the `GodotView#Renderer` to pass the value to the native side when `GodotView#Renderer#onSurfaceCreated` is invoked through `GodotLib#newcontext`.
Or add a `getSurface` method to `Godot.java` that's accessed by `GodotJavaWrapper` on the native side.

Reply:
Yes, I think that last one is the direction we'll go through, adding a getSurface method to Godot.java and adding it to the wrapper. If we can get the other things inside of the GDNative module then we an cancel this PR.

Reply:
owh, no we still need this PR because we do need to return the surface to GDNative :) But access should be through the wrapper :)